cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -356,6 +361,20 @@ protected static ProgramRecord makeProgramRecord(String appId, ProgramSpecificat
     return null;
   }
 
+  protected Map<RunId, ProgramRuntimeService.RuntimeInfo> findAllRuntimeInfos(String namespaceId, String appId,
```

Comment:
actually, should we move this method to ``ProgramRuntimeService``? Instead of passing a ``ProgramRuntimeService`` to a helper method? That way even things that do not use ``AbstractAppFabricHttpHandler``, but need this functionality, can use it on a ``ProgramRuntimeService`` object?